src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -116,15 +110,24 @@ private <T> List<T> runSnapshotBuilderQuery(
 
   public SnapshotBuilderGetConceptsResponse getConceptChildren(
       UUID datasetId, Integer conceptId, AuthenticatedUserRequest userRequest) {
-    // TODO: Build real query - this should get the name and ID from the concept table, the count
-    // from the occurrence table, and the existence of children from the concept_ancestor table.
     Dataset dataset = datasetService.retrieve(datasetId);
-
-    String cloudSpecificSQL = buildConceptChildrenQuery(dataset, conceptId, userRequest);
-
+    String cloudSpecificSQL;
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      DatasetModel datasetModel = datasetService.retrieveDatasetModel(datasetId, userRequest);
+      cloudSpecificSQL =
+          buildConceptChildrenQuery(conceptId, BigQueryVisitor.bqTableName(datasetModel));
+    } else if (cloudPlatformWrapper.isAzure()) {
+      String datasourceName =
+          datasetService.getOrCreateExternalAzureDataSource(dataset, userRequest);
+      cloudSpecificSQL =
+          buildConceptChildrenQuery(conceptId, SynapseVisitor.azureTableName(datasourceName));
+    } else {
+      throw new NotImplementedException("Cloud platform not implemented");
+    }
     List<SnapshotBuilderConcept> concepts =
         runSnapshotBuilderQuery(
-            cloudSpecificSQL,
+            sanitize(cloudSpecificSQL),
```

Comment:
is this the correct way to sanitize the sql before running it? I saw this method used elsewhere for sanitizing files.

Reply:
I think we'll investigate ways to go about that in this spike: https://broadworkbench.atlassian.net/browse/DC-806
But, that will be more of a concern when we're adding free text from the user to the query. For this endpoint, we're only using an integer value from the user, so it should be safe without any further sanitization, so this sanitize() method can be removed.